stage1/usr_from_kvm/lkvm/patches/dead_lock_fix.patch

```diff
@@ -0,0 +1,40 @@
+--- ./build-rkt-1.0.0+git/tmp/usr_from_kvm/lkvm/src/builtin-run.c	2016-02-18 16:50:52.096000000 +0000
++++ ../builtin-run.c	2016-02-18 16:40:30.920000000 +0000
+@@ -634,7 +634,7 @@
+ static int kvm_cmd_run_work(struct kvm *kvm)
+ {
+ 	int i;
+-	void *ret = NULL;
++	int exit_status = 0;
+ 
+ 	for (i = 0; i < kvm->nrcpus; i++) {
+ 		if (pthread_create(&kvm->cpus[i]->thread, NULL, kvm_cpu_thread, kvm->cpus[i]) != 0)
+@@ -642,7 +642,27 @@
+ 	}
+ 
+ 	/* Only VCPU #0 is going to exit by itself when shutting down */
+-	return pthread_join(kvm->cpus[0]->thread, &ret);
++	/* */
++	/* BUT It's not actually true for reboot sequence  */
++	/* Reboot is sending SIGKVMEXIT signal to all cpus */
++	/* and that is causing race condition on removing  */
++	/* devices (which are pausing exiting cpus).       */
++	/* This piece of code waits for death of all       */
++	/* cpu threads   				   */
++	for (i = 0; i < kvm->nrcpus; i++) {
++		int retval;
++		int *ret = 0;
```

Comment:
better set with NULL - not sure it's good idea initialize pointer with 0 - IMHO, that could make segfaults to (write to memory at adress zero).